File: general-interview-prep/javascript/dietPlanPerformance.js
/**
 * @param {number[]} calories
 * @param {number} k
 * @param {number} lower
 * @param {number} upper
 * @return {number}
 */
var dietPlanPerformance = function(calories, k, lower, upper) {
    let sum = calories.slice(0, k).reduce((a, b) => a + b, 0);
    let ans = 0;
    for (let j = k-1; ; ) {
        if (sum < lower) --ans;
        if (sum > upper) ++ans;
        if (++j >= calories.length)
            break;
        sum -= calories[j-k];
        sum += calories[j];
    }
    return ans;
};

File: general-interview-prep/golang/dietPlanPerformance.go
package main

import (
	"fmt"
)

func dietPlanPerformance(calories []int, k int, lower int, upper int) int {
	totalPoints := 0
	sum := 0
	for i := 0; i <= len(calories)-k; i++ {
		for j := i; j < i+k; j++ {
			sum = sum + calories[j]
		}

		if sum >= lower && sum > upper {
			totalPoints++
		} else if sum < lower {
			totalPoints--
		}

		sum = 0
	}

	return totalPoints
}

func main() {
	fmt.Println(dietPlanPerformance([]int{1, 2, 3, 4, 5}, 1, 3, 3))
}
